unitTests/magicbox/QueryProcessorTest.ts

```diff
@@ -66,26 +66,39 @@ export function QueryProcessorTest() {
         const stringArraysInResolvedOrder = [['abcd', 'efgh'], ['ijkl'], ['mnop', 'qrst'], ['uvwx']];
         let stringPromisesInResolvedOrder: (string[] | Promise<string[]>)[];
         let stringPromisesInParameterOrder: (string[] | Promise<string[]>)[];
+        let queryProcessor: QueryProcessor<string>;
+        let returnedPromise: Promise<IQueryProcessResult<string>>;
 
         beforeEach(() => {
           stringPromisesInResolvedOrder = [
-            stringArraysInResolvedOrder[0],
+            wait(resolveTimes[3], stringArraysInResolvedOrder[0]),
             wait(resolveTimes[1], stringArraysInResolvedOrder[1]),
             wait(resolveTimes[2], stringArraysInResolvedOrder[2]),
-            wait(resolveTimes[3], stringArraysInResolvedOrder[3])
+            stringArraysInResolvedOrder[3]
           ];
           stringPromisesInParameterOrder = [
-            stringPromisesInResolvedOrder[2],
-            stringPromisesInResolvedOrder[1],
             stringPromisesInResolvedOrder[0],
+            stringPromisesInResolvedOrder[1],
+            stringPromisesInResolvedOrder[2],
             stringPromisesInResolvedOrder[3]
```

Comment:
Since the indices are now in ascending order, do we still need the variable `stringPromisesInParameterOrder`? Are the two arrays now the same? e.g.

```
stringPromisesInParameterOrder = stringPromisesInResolvedOrder
```